src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -164,6 +164,12 @@ public IEnumerable<DynamicJsonValue> GetAttachmentsMetadataForDocumentWithCounts
                 Debug.Assert(false);// never hit
             }
 
+            var existingDocument = _documentsStorage.Get(context, documentId);
+            if (existingDocument == null)
+            {
+                throw new DocumentDoesNotExistException($"Document {documentId} does not exist. Cannot store attachment to a missing document.");
```

Comment:
There is a `Cannot put attachment {name} on a non existent document '{documentId}'` exception bellow. Wouldn't it be enough?

Reply:
Yes, you right, I'll remove it

Reply:
Can you switch the ex from InvalidOperationEx to DocumentDoesNotExist?

Reply:
Yes